[BE] docs: 리뷰 모아보기 RestDocs 작성

nayonsoso opened the pull request:
<!-- 제목: [BE/FE/All] (기능 등) -->
<!-- 아래에 이슈 번호를 매겨주세요 -->

- resolves #782

---

### 🚀 어떤 기능을 구현했나요 ?
- 오늘 프엔 / 백엔 3대 2로 대결했던 API 문서 회의를 RestDocs 로 만들었습니다.

### 🔥 어떻게 해결했나요 ?
- URI 와 응답 객체를 생성했습니다.
- 컨트롤러와 서비스 코드 뼈대를 만들었습니다.
- RestDocs api test 에서 서비서 코드를 모킹해서 정해진 응답을 반환하게 stub 한 후, 실제 응답과 비교하는 테스트를 작성했습니다.

### 📝 어떤 부분에 집중해서 리뷰해야 할까요?
- 이 PR을 기준으로 이후의 작업이 진행될 것 같으니, uri 나 ResponseDto 이름 같은 것들 만족스럽지 않은 부분들 있으시면 리뷰해주세요. (각자 하다가고치면 컨틀릭 발생🙀)

### 📚 참고 자료, 할 말
- 전체 테스트 실행 -> clean -> build 후 resources -> static 하위에 있는 index.html 로 직접 확인해주세용!

nayonsoso committed:
refactor: 실제와 다른 부분 api docs 에서 수정

- CHECKBOX이면, answer(서술형 답변)이 null 이어야 한다.

nayonsoso committed:
refactor: 함수 이름 변경

- 이제 세션을 사용하는 것이 기본 값이므로, 다른 함수들과 형식을 통일한다.

nayonsoso committed:
feat: 리뷰 요약 응답 dto 생성

nayonsoso committed:
feat: 리뷰 요약 컨트롤러 메서드, 서비스 코드 작성

- 컴파일 에러만 안나게 해둔 상태

nayonsoso committed:
test: 리뷰 요약 API 테스트 작성

nayonsoso committed:
feat: 섹션 목록 응답 dto 생성

nayonsoso committed:
feat: 섹션 목록 조회 컨트롤러, 서비스 코드 작성

nayonsoso committed:
test: 섹션 목록 조회 API 테스트

nayonsoso committed:
docs: RestDocs 에 추가

nayonsoso committed:
feat: 섹션 별 리뷰 응답 dto 생성

nayonsoso committed:
feat: 섹션 별 리뷰 조회 컨트롤러, 서비스 코드 작성

nayonsoso committed:
test: 섹션 별 리뷰 조회 API 테스트

nayonsoso committed:
docs: RestDocs 에 추가

hyeonji1220 commented on backend/src/main/java/reviewme/review/service/dto/response/list/ReceivedReviewSummaryResponse.java:
```diff
@@ -0,0 +1,8 @@
+package reviewme.review.service.dto.response.list;
+
+public record ReceivedReviewSummaryResponse(
```
여러 리뷰들의 요약이라는 의미니 ReceivedReivew's'SummaryResponse는 어떨까요?

donghoony replied:
밖에서 재사용될 걸 생각해서 `ReviewGroupSummary`라고 한다면 너무 방대한 개념일까요?

nayonsoso replied:
기존의 ReceivedReviewSummaryResponse
커비의 ReceivedReivewsSummaryResponse
아루의  ReviewGroupSummaryResponse
모두 좋은 의견들이네요~

1. Review vs Group
이 response dto 에 들어가는 값은 [리뷰이 이름 / 프로젝트 이름 / 받은 리뷰 전체 개수] 예요. 이 중, 리뷰이 이름 & 프로젝트 이름은 리뷰 그룹에 해당하고 받은 리뷰 전체 개수는 리뷰에 해당하는 정보라서 생각이 갈리는 것 같아요.
그런데 사용되는 방식을 보면, '❗️어떤 이름으로, 어떤 프로젝트에 대해서 받은 리뷰❗️가 몇개인지'를 알려주기 위해 사용되고 있어요. 이런 맥락을 봤을 때 저는 리뷰에 대한 정보에 더 가깝다고 생각해요!

2. Reviews vs Review
리뷰들에 대한 요약이니 Reviews 가 더 적절하다는 커비 의견에 설득되었습니당~ 반영할게요!

hyeonji1220 commented on backend/src/main/java/reviewme/template/service/dto/response/SectionNameResponse.java:
```diff
@@ -0,0 +1,7 @@
+package reviewme.template.service.dto.response;
+
+public record SectionNameResponse(
+        String name,
+        long id
```
다른 reponse와 동일하게 id가 위에 있는 건 어떨까요?

nayonsoso replied:
반영했습니다!

hyeonji1220 commented on backend/src/docs/asciidoc/index.adoc:
review-gather.adoc 도 추가해야겠네요!

nayonsoso replied:
추가했습니다😊 https://github.com/woowacourse-teams/2024-review-me/pull/785/commits/6691adbf39a7e3df08f231d97d7f9d596fc84ccf

hyeonji1220 commented on backend/src/main/java/reviewme/review/service/dto/response/gathered/QuestionResponse.java:
QuestionReponse라는 동일명의 response가 다른 패키지에 이미 존재해서 문제가 없으려나 생각이 들었는데요,
reponse는 재사용하지 않는다는 합의가 있었으니 위치한 패키지를 잘 보고 쓰면 되는 걸까요? 아니면 네이밍에 좀 더 의미를 담는 게 좋을까요?

nayonsoso replied:
동일한 이름의 클래스가 있으면 헷갈릴 것 같긴 한데요😅
그렇다고 '다른 패키지에 같은 이름의 패키지가 있으니 이름을 바꿔야지~'는 아닌 것 같습니다. 클래스 이름 지을 때 외부 요인이 이유가 되면 안될 것 같아요.
패키지 잘 보고 & 인자 잘 보고 하면 된다 생각해요!

donghoony replied:
이건 다르게 명명하는 게 낫겠습니다, 같은 이름의 클래스를 하나의 프로젝트에서 사용하는 경우를 본 적이 없어서요. `@Transactional`같은 경우 `jakarta`와 `jpa`에 같은 이름으로 있지만 서로 결이 다른 것처럼요.

nayonsoso replied:
추가)
아루의 오프라인 피드백 "패키지와 클래스 이름이 독립적이어야 한다는 말도 맞지만, 하나의 어플리케이션에서는 같은 이름의 클래스가 있다면 헷갈릴 수 있으니 이름 바꾸는게 좋을 것 같다"를 받고 SimpleQuestionResponse 로 변경했습니다~

hyeonji1220 commented on backend/src/main/java/reviewme/review/service/dto/response/gathered/GatheredReviewResponse.java:
```diff
@@ -0,0 +1,15 @@
+import jakarta.annotation.Nullable;
+import java.util.List;
+
+public record GatheredReviewResponse(
```
해당 응답은 섹션 내의 질문별 응답이네요!
그렇다면 QuestionReviewResponse 이런 네이밍은 어떨까요?

nayonsoso replied:
의미를 더 전달해야 한다는 것 동의합니다.
그래서 이렇게 바꿔봤어요,

- GatheredReviewsResponse ➡️ ReviewsGatheredBySectionResponse
- GatheredReviewResponse ➡️ 
ReviewsGatheredByQuestionResponse

hyeonji1220 reviewed (changes requested):
어떻게 구현할지 백엔드끼리도 합의되고 api 문서 작성 먼저 역시 좋네요!! 
회의부터 문서까지 수고했어요~~
I left 소소쓰 질문쓰~~

+ URL에 대해서는 같이 얘기해봐요!

nayonsoso committed:
refactor: dto 이름 변경

- 복수의 리뷰들이 대한 이름이므로, s를 붙이도록

nayonsoso committed:
refactor: 인자 순서 변경

nayonsoso committed:
refactor: 응답 dto 이름 변경

- 하나의 모듈에는 되도록 같은 이름의 클래스가 없도록

nayonsoso committed:
refactor: 응답 dto 이름 변경

- GatheredReviewsResponse -> ReviewsGatherBySectionResponse
- GatheredReviewResponse ->
ReviewsGatherByQuestionResponse

nayonsoso committed:
fix: API 테스트 수정

- DTO 인자 순서 변경 반영

nayonsoso committed:
refactor: Nested 클래스 이름 영어로 변경

nayonsoso committed:
refactor: Nested 클래스 이름 대문자로 변경

nayonsoso committed:
docs: 쿠키 설명 변경

- 세션 쿠키 -> 세션 ID

nayonsoso committed:
docs: 문서 구조 변경

- 모이보기 파트 따로 분리

nayonsoso committed:
refactor: dto 이름 수정

- GatherBy -> GatheredBy

donghoony reviewed (approved):
오프라인으로 이야기했습니다~!
주로 이야기한 내용으로는:
- `@Nested` 클래스명에 Non-ascii 문자가 들어가면 빌드가 안된다.
  - `@DisplayName`을 `@Nested`에 한해 사용한다.
- 문서 가독성 및 구조화
- Dto 네이밍

입니다~

nayonsoso committed:
docs: 응답 내용에 대한 설명 보충

Kimprodp reviewed (approved):
기대가 됩니다
